.github/workflows/api.yml

```diff
@@ -0,0 +1,118 @@
+name: API Compatibility
+
+on:
+  schedule:
+    - cron: "0 12 * * *"
+
+env:
+  BAZEL_OPTIMIZATION: --copt=-msse4.2 --copt=-mavx --compilation_mode=opt
+
+jobs:
+
+  macos:
+    name: macOS ${{ matrix.python }} + ${{ matrix.version }}
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        python: ['3.8']
+        version: ['tensorflow==2.4.0rc4:tensorflow-io-nightly', 'tf-nightly:tensorflow-io-nightly']
+    steps:
+      - uses: actions/checkout@v2
+      - uses: docker-practice/actions-setup-docker@v1
```

Comment:
@yongtang last time we checked, docker on macOS was a bit unstable. Did they fix the issues in their latest releases? If yes, then we can use it to test additional datasets in macOS env of `build.yml` as well.

Reply:
@kvignesh1420 From the experience it looks like docker on GitHub Actions' macOS is much more stable. We can gradually re-enable the docker on macOS for dataset tests. It is a little slow compared with on Ubuntu though.